src/frontend/components/StudentLiveAdvertising/StudentLiveRegistration/RegistrationForm/index.spec.tsx

```diff
@@ -0,0 +1,212 @@
+
+import { RegistrationForm } from '.';
+
+const setValues = jest.fn();
+const setRegistrationCompleted = jest.fn();
+
+let matchMedia: MatchMediaMock;
+
+jest.mock('data/sideEffects/createLiveRegistration', () => ({
+  createLiveRegistration: jest.fn(),
+}));
+const mockCreateLiveRegistration =
+  createLiveRegistration as jest.MockedFunction<typeof createLiveRegistration>;
+
+describe('<RegistrationForm />', () => {
+  beforeEach(() => {
+    matchMedia = new MatchMediaMock();
+    jest.clearAllMocks();
+
+    /*
+      make sure to remove all body children, grommet layer gets rendered twice, known issue
+      https://github.com/grommet/grommet/issues/5200
+    */
+    document.body.innerHTML = '';
+    document.body.appendChild(document.createElement('div'));
+  });
+
+  afterEach(() => {
+    matchMedia.clear();
+  });
+
+  it('renders the form without values', () => {
+    const values = {
+      email: undefined,
+    };
+
+    render(
+      wrapInIntlProvider(
+        <RegistrationForm
+          values={values}
+          setValues={setValues}
+          setRegistrationCompleted={setRegistrationCompleted}
+        />,
+      ),
+    );
+
+    screen.getByRole('heading', { name: 'Email address' });
+    screen.getByRole('textbox', { name: '' });
+    screen.getByRole('button', { name: 'Register' });
+    expect(
+      screen.queryByText('You have to submit a valid email to register.'),
+    ).not.toBeInTheDocument();
+  });
+
+  it('renders the form with initial value', () => {
+    const values = {
+      email: 'some.email@openfun.fr',
+    };
+
+    render(
+      wrapInIntlProvider(
+        <RegistrationForm
+          values={values}
+          setValues={setValues}
+          setRegistrationCompleted={setRegistrationCompleted}
+        />,
+      ),
+    );
+
+    screen.getByRole('heading', { name: 'Email address' });
+    screen.getByRole('textbox', { name: '' });
+    screen.getByDisplayValue('some.email@openfun.fr');
+    screen.getByRole('button', { name: 'Register' });
+    expect(
+      screen.queryByText('You have to submit a valid email to register.'),
+    ).not.toBeInTheDocument();
+  });
+
+  it('calls parent on input change', () => {
+    const values = {
+      email: 'some.email@openfun.co',
+    };
+
+    render(
+      wrapInIntlProvider(
+        <RegistrationForm
+          values={values}
+          setValues={setValues}
+          setRegistrationCompleted={setRegistrationCompleted}
+        />,
+      ),
+    );
+
+    screen.getByRole('heading', { name: 'Email address' });
+    screen.getByRole('textbox', { name: '' });
+    screen.getByDisplayValue('some.email@openfun.co');
+    screen.getByRole('button', { name: 'Register' });
+    expect(
+      screen.queryByText('You have to submit a valid email to register.'),
+    ).not.toBeInTheDocument();
+
+    fireEvent.change(screen.getByRole('textbox', { name: '' }), {
+      target: { value: 'an.other.email@openfun.fr' },
+    });
```

Comment:
Wouldn't it be better to use userEvent ?

https://testing-library.com/docs/user-event/intro#difference-to-fireevent